### DISCUSSION TITLE
Hardware i2s pdm tx

### DISCUSSION THREAD

**Discussion by Jeronimo59** [pschatzmann/arduino-audio-tools#1492] [Q&A] [answered]
Hello, and many thanks for this outstanding work !
I've found that ESP32 C3 & S3 now have Hardware i2s driver with a pdm mode.
https://docs.espressif.com/projects/esp-idf/en/v5.1/esp32/api-reference/peripherals/i2s.html
Are you considering to integrate this use case in your toolbox ?

Meanwhile, I am struggling with pdm tx in software, and didn't find any example (the only pdm example I found is for pdm RX).
Did I miss it ?

Thanks again,

Jérôme

**Comment by pschatzmann** [comment, marked as answer]
It is already integrated, but I did not have the chance to test it yet. Just set the signal type in the [I2S configuration](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_i2_s_config_e_s_p32.html) and use the ESP32 v3.0.0 core

Any help will be appreciated.

**Reply by Jeronimo59** [reply]
Thanks for your prompt response.

I did not realize your implementation of PDM used ESP32 hardware driver already (I thought it was software-based).
I will test & get back to you.

Jerome
